apollo-federation/src/composition/satisfiability/satisfiability_error.rs

```diff
@@ -0,0 +1,466 @@
+use apollo_compiler::Node;
+use apollo_compiler::ast;
+use apollo_compiler::collections::IndexSet;
+use apollo_compiler::executable;
+use multimap::MultiMap;
+use petgraph::graph::EdgeIndex;
+
+use crate::bail;
+use crate::error::FederationError;
+use crate::error::SingleFederationError;
+use crate::operation::FieldSelection;
+use crate::operation::InlineFragment;
+use crate::operation::InlineFragmentSelection;
+use crate::operation::Operation;
+use crate::operation::SelectionId;
+use crate::operation::SelectionSet;
+use crate::query_graph::QueryGraphEdgeTransition;
+use crate::query_graph::QueryGraphNodeType;
+use crate::query_graph::graph_path::TransitionGraphPath;
+use crate::query_graph::graph_path::Unadvanceables;
+use crate::schema::ValidFederationSchema;
+use crate::schema::position::CompositeTypeDefinitionPosition;
+use crate::schema::position::FieldDefinitionPosition;
+use crate::schema::position::TypeDefinitionPosition;
+
+/// Returns a satisfiability error in Ok case; Otherwise, returns another error in Err case.
+#[allow(unused)]
+fn satisfiability_error(
+    unsatisfiable_path: &TransitionGraphPath,
+    _subgraphs_paths: &[TransitionGraphPath],
+    subgraphs_paths_unadvanceables: &[Unadvanceables],
+) -> Result<SingleFederationError, FederationError> {
```

Comment:
nit: it looks weird to return `Result<Error, Error>`....

Reply:
I know. But, the function's successful output is an error value, while it may fail with another error 🤣

Reply:
(Echoing some feedback from a meeting) It was suggested here that having an argument to receive the `SingleFederationError` (and having the function return `Result<(), FederationError>`) may be easier to understand, at least until we split out composition errors into a separately named error.